### ISSUE TITLE
app-compat: Assertion failed when running `node` with `app-elfloader`

### ISSUE THREAD

**Issue by razvand** [unikraft/unikraft#859]
### Describe the bug

When using [`app-elfloader`](https://github.com/unikraft/app-elfloader) (via the `run_app.sh` script in [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader)) to run the [`node-musl` static PIE build](https://github.com/unikraft/static-pie-apps/tree/master/node), an assertion appears.

### Steps to reproduce

You can use the [`./do.sh`](https://github.com/unikraft-upb/scripts/tree/main/make-based/app-elfloader) script in the [`unikraft-upb/scripts` repository](https://github.com/unikraft-upb/scripts):

```console
$ ./do.sh setup
$ ./do.sh build
$ ./do.sh run_built node_static_musl
```

### Expected behavior

A Node server should start, waiting for connections.

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

### Relevant log output

```shell
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, fd:-1, 0) = va:0x1000093000
eventfd2(0x0, 0x80800, ...) = Invalid argument (-22)
close(fd:6) = OK
close(fd:7) = OK
munmap(va:0x1000093000) = OK
getpid() = pid:0
../app-elfloader/build/app-elfloader_kvm-x86_64[0]: ../src/tracing/agent.cc:55:node::tracing::Agent::Agent()writev(0x2, 0x40008f790, ...) = 0x6c
: Assertion `(uv_loop_init(&tracing_loop_)) == (0)' failed.
writev(0x2, 0x40008f870, ...) = 0x3c
rt_sigprocmask(0x0, 0x474497888, ...) = 0x0
tkill(0x0, 0x6, ...) = 0x0
rt_sigprocmask(0x2, 0x40008fa40, ...) = 0x0
rt_sigprocmask(0x0, 0x474497890, ...) = 0x0
rt_sigaction(0x6, 0x40008faf0, ...) = 0x0
tkill(0x0, 0x6, ...) = 0x0
rt_sigprocmask(0x1, 0x40008fae8, ...) = 0x0
```

**Comment by flpostolache** [comment]
Hello, @razvand. I recreated the problem using the steps above. This specific problem comes from the failure of the `eventfd2` syscall. These three PRs #754 #756 #798 added on top of Unikraft solve this issue. For the syscall to succeed only #754 is needed.

**Comment by flpostolache** [comment]
Update. I wasn't aware of PR #850. Because `9pfs_ioctl` is not implemented, the `fcntl` function will fail and the assert in `do_eventfd` will stop the execution of the program.

**Comment by razvand** [comment]
So, @maniatro111, what is the state of it? I moved it for release 0.14 just in case.

**Comment by maniatro111** [comment]
Still working on it.

**Comment by razvand** [comment]
Solved in #850
